code-challenges.test.js

```diff
@@ -16,47 +16,272 @@
 
 // a) Create a test with expect statements using the variable provided.
 
-var hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-var hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-var hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
+//JEST TEST: testing function fullHouse that will take an array, and determine if it is full house (1 pair and 1 three of a kind)
+describe("testing function fullHouse", () => {
+    it("should take an array and check whether there is exactly one pair and one three of a kind", () => {
+
+        var hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        var hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        var hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
 
+      expect(fullHouse(hand1)).toEqual(true)
+      expect(fullHouse(hand2)).toEqual(false)
+      expect(fullHouse(hand3)).toEqual(false)
+    })
+  })
+//test fails because: ReferenceError: fullHouse is not defined
 
 
 // b) Create the function that makes the test pass.
+//NOTES: FULL HOUSE CONDITION: 3 of a kind AND 2 of a kind
+//WE immediately know that we can only have two kinds of elements. 
+//if a set has length of greather than two => we don't have a full house
+//WE also need to check that ratio is 2 to 3 of only 1 number
+
+
+//create function fullHouse that takes an array of length 5
+// const fullHouse = (array) => {
+//     //create a new array that will filter out the duplicates
+//     let numbers = new Set(array);
+//     // console.log(array);
+//     // console.log(numbers);
+
+//     //counter for a number
+//     let countNum1 = 0;
+
+//     //check if there is only 1 element or more than 2 elements in the array
+//     //if true, early return false since a full house would be impossible
+//     // console.log(numbers.size);
+//     if (numbers.size != 2){
+//         return false;
+//     } 
+ 
+//     //iterate through array
+//     for (let i = 0; i < array.length; i++) {
+//         //knowing that there are only 2 numbers, we only need to keep track of 1 so I choose 1st number in array
+//         //if current number in array = 1st number in array then countNum1++
+//         if(array[i] === array[0]){
+//             countNum1++;
+//         }
+//     } 
+
+//     //if count is two or three, then the other number must occur three or two times respectively
+//     //meaning there is a full house and we should return true
+//     // console.log(countNum1);
+//     if(countNum1 === 2 || countNum1 ===3){
+//         return true;
+//     }
+    
+//     return false;
+
+// }
+
+//ran test, failed because numbers.length does not work. numbers is now a set and the correct method should be .size to check the number of unique elements
+//also initially tried comparing array[i] to numbers[0] but it doesn't seem like you can access set items that way. 
+
+//====>Question for Austin, any easy way to pull a value from a set similar to how we index array elements (i.e. array[0])?
```

Comment:
Set.value() which gives you all the values in the set as an array. So maybe the .find mehtod works here (save for the fact that it will give you back the first occurance)
so maybe something like 

value1 = SetName.values().find(element).pop() so your set is reduced by the first occurrence of the element

I'd have to look deeper into this

Reply:
I can think of 2 issues with my previous idea-- still I'll leave it because it's kinda helpful